include/mbgl/style/filter.hpp

```diff
@@ -35,6 +37,16 @@ class Filter {
     friend bool operator!=(const Filter& lhs, const Filter& rhs) {
         return !(lhs == rhs);
     }
+    
+    mbgl::Value serialize() const {
```

Comment:
Should this be used in place of https://github.com/mapbox/mapbox-gl-native/blob/master/src/mbgl/style/conversion/stringify.hpp#L131 (and tested via that code path, too)?

Reply:
`stringify` is currently used only in generating the `layoutKey` for grouping layers. Is it being considered for use in any other areas? I think adding more support for the legacy filters is unnecessary. The converted expressions are sufficient for generating keys.  Lighter footprint left by the legacy filters means we can deprecate them more easily in the future.

Reply:
Oh, huh...  So where is this `serialize()` being used?  It looks like it's being introduced in this PR, but I don't see any corresponding use of it being added except for in the unit test